torch_geometric/nn/conv/Bayesian_GCNConv.py

```diff
@@ -0,0 +1,228 @@
+from typing import Optional
+
+import torch
+import torch.nn.functional as F
+from torch import Tensor
+from torch.nn import Parameter
+from torch_sparse import SparseTensor, mul
+from torch_sparse import sum as sparsesum
+
+from torch_geometric.nn.conv import MessagePassing
+from torch_geometric.typing import Adj, OptPairTensor, OptTensor
+# for the last version of torch_geometric
+from torch_geometric.utils import scatter, spmm
+from torch_geometric.utils.num_nodes import maybe_num_nodes
+
+# Inspired from IntelLabs :
+# https://github.com/IntelLabs/bayesian-torch/blob/main/bayesian_torch/
+# layers/base_variational_layer.py
+
+
+def gcn_norm(edge_index, edge_weight=None, num_nodes=None, dtype=None):
+
+    if isinstance(edge_index, SparseTensor):
+        adj_t = edge_index
+        if not adj_t.has_value():
+            adj_t = adj_t.fill_value(1., dtype=dtype)
+        deg = sparsesum(adj_t, dim=1)
+        deg_inv_sqrt = deg.pow_(-0.5)
+        deg_inv_sqrt.masked_fill_(deg_inv_sqrt == float('inf'), 0.)
+        adj_t = mul(adj_t, deg_inv_sqrt.view(-1, 1))
+        adj_t = mul(adj_t, deg_inv_sqrt.view(1, -1))
+        return adj_t
+
+    else:
+        num_nodes = maybe_num_nodes(edge_index, num_nodes)
+
+        if edge_weight is None:
+            edge_weight = torch.ones((edge_index.size(1), ), dtype=dtype,
+                                     device=edge_index.device)
+
+        row, col = edge_index[0], edge_index[1]
+        deg = scatter(edge_weight, col, dim=0, dim_size=num_nodes)
+        deg_inv_sqrt = deg.pow_(-0.5)
+        deg_inv_sqrt.masked_fill_(deg_inv_sqrt == float('inf'), 0)
+        return edge_index, deg_inv_sqrt[row] * edge_weight * deg_inv_sqrt[col]
+
+
+class BGCNConv(MessagePassing):
```

Comment:
Please also add some tests for `BGCNConv`.

Reply:
With pleasure, what kind of test are you thinking about ? We can use a graph from pyg datasets and test it with this new conv layer ?

Reply:
Just test the basic functionality of `BGCNConv`, you may refer https://github.com/pyg-team/pytorch_geometric/blob/master/test/nn/conv/test_sg_conv.py for a simple example :)

Reply:
ok will do that tomorrow morning haha, I am so sleepy rn